pbm/pbm_util.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pbm
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/vmware/govmomi/pbm/types"
+)
+
+// A struct to capture pbm create spec details.
+type CapabilityProfileCreateSpec struct {
+	Name           string
+	Description    string
+	Category       string
+	CapabilityList []Capability
+}
+
+// A struct to capture pbm capability instance details.
+type Capability struct {
+	ID           string
+	Namespace    string
+	PropertyList []Property
+}
+
+// A struct to capture pbm property instance details.
+type Property struct {
+	ID       string
+	Operator string
+	Value    string
+	DataType string
+}
+
+func CreatePbmCapabilityProfileSpec(pbmCreateSpec CapabilityProfileCreateSpec) (types.PbmCapabilityProfileCreateSpec, error) {
+	var pbmCapabilityProfileSpec types.PbmCapabilityProfileCreateSpec
+	capabilities, err := createCapabilityInstances(pbmCreateSpec.CapabilityList)
+	if err != nil {
+		return pbmCapabilityProfileSpec, err
```

Comment:
Change method return type to `*types.PbmCapabilityProfileCreateSpec` and return `nil` here.